document/introduction/index.rst

```diff
@@ -1,6 +1,54 @@
 Introduction
 ============
 
-.. todo::
+WebAssembly (abbreviated Wasm [#wasm]_) is a *safe, portable, low-level code format*
+designed for efficient execution and compact representation.
+Its main goal is to enable high performance applications on the Web, but it does not make any Web-specific assumptions or provide Web-specific features, so can be employed in other environments as well.
 
-   All there is to say
+WebAssembly is an open standard developed by a `W3C Community Group <https://www.w3.org/community/webassembly/>`_ that includes representatives of all major browser vendors.
+
+This document describes version |release| of the standard.
+It is intended that it will be superseded by new incremental releases with additional features in the future.
+
+
+Design Goals
+------------
+
+.. index:: design goals, portability
+
+The design goals of WebAssembly are the following:
+
+* Fast, safe, and portable *semantics*:
+
+  * **Fast**: executes with near native code performance, taking advantage of capabilities common to all contemporary hardware.
+
+  * **Safe**: code is validated and executes in a memory-safe, sandboxed environment preventing data corruption or security breaches.
```

Comment:
The idiom of using linear memory as an address space is fundamentally "unsafe" because there's no bounds checking for arrays allocated within the address space, or protections for "deallocated" memory (in a malloc+free sense). At one level, this isn't WebAssembly's problem, however this use case is the main reason why WebAssembly has linear memory, so it's a little unfriendly to ignore it. Corrupted linear memory can participate in some forms of security breaches, even though it's sandboxed from the underlying browser/OS.

It's obviously a matter of perspective. From my perspective, just saying "sandboxed" is sufficient and avoids confusion.